packages/kbn-legacy-logging/src/utils/get_payload_size.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import type { ReadStream } from 'fs';
+import type { ResponseObject } from '@hapi/hapi';
+
+/**
+ * Attempts to determine the size (in bytes) of a hapi/good
+ * responsePayload based on the payload type. Falls back to
+ * `undefined` if the size cannot be determined.
+ *
+ * This is similar to the implementation in `core/server/http/logging`,
+ * however it uses more duck typing as we do not have access to the
+ * entire hapi request object like we do in the HttpServer.
```

Comment:
I went back and forth on whether to try to share the http server code for handling this with legacy logging, and ended up duplicating some of the code for two reasons:
1. There wasn't a logical place to put the code; it would need to be in a package to share it, but it didn't really make sense in `@kbn/logging`, as it needs to be aware of the hapi response structure and only really has to do with the http server.
2. It turns out `@hapi/good` is only including a subset of the hapi response object in the payload anyway, so in order to write code that can be shared we'd need to "dumb down" the http server implementation a bit.